src/main.rs

```diff
@@ -0,0 +1,223 @@
+use libc::{uid_t, gid_t};
+use std::io::prelude::*;
+use std::ffi::CString;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::process::exit;
+
+struct BindMount {
+	dest: PathBuf,
+	src: PathBuf
+}
+
+struct Config {
+	rootfs: PathBuf,
+	uid: uid_t,
+	gid: gid_t,
+	args: Vec<String>,
+	bind_mounts: Vec<BindMount>,
+}
+
+// TODO: This function does not really perform error checking;
+//       for now, we assume that xbstrap passes sane values.
+fn make_config_from_cli() -> Config {
+	let matches = clap::App::new("cbuildrt")
+		.arg(clap::Arg::with_name("cbuild-json")
+			.help("cbuild.json file")
+			.required(true)
+		)
+		.get_matches();
+
+	let cfg_f = File::open(matches.value_of("cbuild-json").unwrap())
+		.expect("unable to open cbuild.json");
+	let cfg : serde_json::Value = serde_json::from_reader(cfg_f)
+		.expect("failed to parse cbuild.json");
+
+	return Config{
```

Comment:
you could use [`serde_derive`](https://serde.rs/derive.html) instead of doing this manually